src/joint_types.jl

```diff
@@ -131,6 +131,80 @@ function velocity_to_configuration_derivative!(q̇::AbstractVector, jt::Quaterni
     nothing
 end
 
+
+"""
+$(SIGNATURES)
+
+Copy the matrix ``source`` to a block of the same size in ``destination``
+spanning the given given range of rows and columns. In Julia v0.7 and above we
+should be able to eliminate this function in favor of:
+
+```julia
+destination[dest_rows, dest_cols] .= source
+```
+
+since the resulting view will be elided.
+"""
+function _copyto_offset!(destination::AbstractMatrix, source::AbstractMatrix, dest_rows::UnitRange{Int} = 1:size(source, 1), dest_cols::UnitRange{Int} = 1:size(source, 2))
+    @boundscheck length(dest_rows) == size(source, 1)
+    @boundscheck length(dest_cols) == size(source, 2)
+    for j in 1:size(source, 2)
+        for i in 1:size(source, 1)
+            destination[dest_rows[i], dest_cols[j]] = source[i, j]
+        end
+    end
+end
+
+
+"""
+$(SIGNATURES)
+
+Compute the jacobian ``Q_v`` which maps joint velocity to configuration
+derivative for the joint type ``jt``:
+
+```math
+\\dot{q} = Q_v v
+```
+
+This updates the entries of Q_v in place. Note that only the structurally
+non-zero entries in V_q are modified, so you may need to initialize V_q
```

Comment:
```
`Q_v`
```

instead of V_q I guess?

Reply:
Thought about this a bit more, and I think that unless it makes things harder elsewhere, I'd prefer to zero the 'non-structurally zero'  parts as well. I'm afraid of subtle bugs going forward, and sparse matrices already handle not explicitly storing zeros in `setindex!`:

```julia
julia> A = spzeros(3, 3)
3×3 SparseMatrixCSC{Float64,Int64} with 0 stored entries

julia> A[2, 2] = 0
0

julia> nnz(A)
0
```